### ISSUE TITLE
Plugin does not pick up coordinates anymore

### ISSUE THREAD

**Issue by almereyda** [fedwiki/wiki-plugin-map#19]
I have [recently](https://hub.docker.com/u/federatedwiki/) [updated](https://github.com/federated-wiki?utf8=%E2%9C%93&query=-wiki-) one of my wikifarms to the latest version.
Unfotunately the map plugin does not pick up the locations anymore.

See a freshly loaded page:

[![bildschirmfoto vom 2016-08-01 00-28-28](https://cloud.githubusercontent.com/assets/1645308/17279804/e7576994-577e-11e6-893e-66764b96a3ca.png)](http://2016.wsf.federated.wiki/view/welcome-visitors/view/commons-space)

See the JSON next to an already a bit zoomed-out version of a map plugin digesting a remote page and also centred somewhere New York'ish.

[![bildschirmfoto von 2016-08-01 00-29-37](https://cloud.githubusercontent.com/assets/1645308/17279813/380bab5c-577f-11e6-840f-6d3d5f599f6e.png)](http://jon.2016.wsf.federated.wiki/view/welcome-visitors/view/a-letter-of-enquiry/2016.wsf.federated.wiki/commons-space)

```
"story": [
    {
      "type": "paragraph",
      "id": "ee38f56d4dd8768a",
      "text": "Commons Space is based at [http://www.openstreetmap.org/node/3313351996 ECTO co-working cooperative, 936 Avenue Mont-Royal Est], 2nd floor. A number of activities, workshops, debates will take place there, but other activities will also be happening in the Forum itself and around the city. "
    },
    {
      "type": "paragraph",
      "id": "fd10ed9e03f031bd",
      "text": "🔗 [http://wiki.remixthecommons.org/images/Program_Commons_Space.pdf Programme] via [https://commonsspace.hackpad.com/Welcome-to-Commons-Space-Cq9t8rdsRd1 Welcome to Commons Space!]"
    },
    {
      "type": "map",
      "id": "d0516adb2b1f7c62",
      "text": "ecto co-working, montréal",
      "latlng": {
        "lat": 45.52696423625801,
        "lng": -73.58018159866333
      },
      "zoom": 18
    },
    {
      "type": "paragraph",
      "id": "a9d99f8b98a9668e",
      "text": "Commons Space is a place for experimentation, exchange and construction  of commons based alternatives to the current economic model.  This space will welcome and support the strategic process of  convergence of commoners and social movements  throughout the WSF. ([https://commonsspace.hackpad.com/Commons-Space-k6rOCvUgyhC welcome pad])"
    }
  ],
```

The coordinates from the `story` actually represent those which were intended (and used to work earlier):

> ### Knoten: Ecto (3313351996)
Écartement des numéros de rues

> Bearbeitet vor mehr als ein Jahr von Peuc
Version 2 · Änderungssatz #28434428
**Standort: 45,5268649, -73,5800498**

[![bildschirmfoto vom 2016-08-01 00-36-29](https://cloud.githubusercontent.com/assets/1645308/17279849/2059033c-5780-11e6-8b2d-e232974f2b36.png)](http://www.openstreetmap.org/node/3313351996)

**Comment by WardCunningham** [comment]
The improved map plugin should honor the item json properties latlng and zoom. The improvements consist of more conventional markup editing and keywords for setting the map boundary without referring to zoom numbers. When specified, these override the item fields.

```
    {
      "type": "map",
      "id": "d0516adb2b1f7c62",
      "text": "ecto co-working, montréal",
      "latlng": {
        "lat": 45.52696423625801,
        "lng": -73.58018159866333
      },
      "zoom": 18
    },
```

I'm getting good results from a recently installed map. Maybe the next step is to look at exact version you have installed using `wiki -v`.

![image](https://cloud.githubusercontent.com/assets/12127/17279918/f9ba1070-5736-11e6-944f-3323ae161f8a.png)

**Comment by paul90** [comment]
So, the map should be displaying a location in Montréral. Not too sure how though, as the `latlng` in the mark-up is in New York!

Ecto gives a location of 45.5244476,-73.5806311
The mark-up has 45.52696423625801,-73.58018159866333

Looking at the page's history I can see the map being added and editted, but at no time does the `latlng` contain anything other than a location in New York.

N.B. The site is not claimed, so in looking at this I have inadvertently edited the page and added a pin in the correct location.

**Comment by almereyda** [comment]
Thanks for your investigations. The map now displays Montréal.
Yet even in the screenshot as seen from `ward.asia.wiki.org` the location displays New York, even though 45.52696423625801,-73.58018159866333 displays in Montréal for me in OSM and Google Maps.

Usually I edit a map plugin and move the viewport to the desired location and zoom. This is reflected by the activity below:

```
{
      "type": "edit",
      "id": "d0516adb2b1f7c62",
      "item": {
        "type": "map",
        "id": "d0516adb2b1f7c62",
        "text": "ecto co-working, montréal"
      },
      "date": 1469618050392
    },
    {
      "type": "edit",
      "id": "d0516adb2b1f7c62",
      "item": {
        "type": "map",
        "id": "d0516adb2b1f7c62",
        "text": "ecto co-working, montréal",
        "latlng": {
          "lat": 45.65820764745017,
          "lng": -73.52806091308594
        },
        "zoom": 11
      },
      "date": 1469618094622
    },
```
Maybe I ran into [another race condition](https://github.com/fedwiki/wiki-plugin-map/issues/16#issue-165133135) which didn't sync the `story` with the `journal` at some point? (Now irrevertably unknowledgeable.)

I will have to replicate my initial issue to have it confirmed.

**Comment by paul90** [comment]
Looking at this again. I have an old version of the plugin installed locally, and it is displaying the location correctly.

So, the latest version of the code fails to pick-up the `latlng` in the JSON.

**Comment by WardCunningham** [comment]
I will check my local repo. I've been working in this area and may have crossed some branches.

**Comment by paul90** [comment]
Looks like a problem with the case of a variable being changed, in the [code](https://github.com/fedwiki/wiki-plugin-map/blob/master/client/map.coffee#L152) it is `item.latLng` but the JSON has `latlng`.

**Comment by WardCunningham** [comment]
Good catch.

**Comment by WardCunningham** [comment]
Here from version 0.2.3 from a year ago.
https://github.com/fedwiki/wiki-plugin-map/blob/14ebc30218353250c1aec3739a5e981c731ba1cc/client/map.coffee#L77

This shows lower case is correct. I believe I am the culprit here and will add a fix to a pull request I have in flight.

**Comment by paul90** [comment]
We also need correct the "about map properties" page that also says `item.latLng`

**Comment by WardCunningham** [comment]
Yes. I'm going to document latlng and accept either.

**Comment by paul90** [comment]
fix included in + wiki-plugin-map@0.3.3
